src/LevelNFT.sol

```diff
@@ -1,51 +1,56 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.13;
+
 import "./ILevelChecker.sol";
 import "solmate/tokens/ERC721.sol";
 import "openzeppelin-contracts/contracts/access/Ownable.sol";
 import "base64/base64.sol";
 
 contract LevelNFT is ERC721, Ownable {
     // Level numbers auto-increment as they are added by the contract owner.
-    uint latestLevel;
+    uint256 public latestLevel;
 
     // Mapping from level numbers to addresses of their implementations.
-    mapping (uint => ILevelChecker) public levels;
+    mapping (uint256 => ILevelChecker) public levels;
 
     // Mapping from level number to how many points it's worth to complete it.
-    mapping (uint => uint) public levelValues;
+    mapping (uint256 => uint256) public levelValues;
     
     // Keeps track of the levels beaten by a given tokenId, while owned by a specific address.
     // This allows a previous owner of a token ID to resume if they get the token back.
-    mapping (uint256 => mapping (uint => bool)) public levelsBeatenByTokenKey;
+    mapping (bytes32 => mapping (uint256 => bool)) public levelsBeatenByTokenKey;
 
-    // The current score of each token ID, based on the current owner and how many levels they've beaten with it.
-    mapping (uint256 => uint) public scoreByTokenId;
+    // The current score of each key, how many levels a token ID and owner pair has beaten.
+    mapping (bytes32 => uint256) public scoreByKey;
```

Comment:
I'm very down for this function, but there is something kinda nice about having scoreByTokenId available. Makes it so that most applications wouldn't have to think about the whole key thing.

Reply:
Perhaps we accept your change, but also add a view function `getScoreByTokenId` which reads the current owner and then looks it up in this mapping.

Reply:
ah i see, you already have this. great.

Reply:
I added `getScore()`, although it's an internal function right now.
Would need to change it to `external` fyi!